From e46c53c21291f7844c72ce964f1eaaa0397abbe2 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Fri, 12 Dec 2014 14:11:04 +0100
Subject: Bug 741434 - Use-after-free after error in GPG signature verification


Index: evolution-data-server-3.12.9~git20141128.5242b0/camel/camel-gpg-context.c
===================================================================
--- evolution-data-server-3.12.9~git20141128.5242b0.orig/camel/camel-gpg-context.c
+++ evolution-data-server-3.12.9~git20141128.5242b0/camel/camel-gpg-context.c
@@ -1889,6 +1889,7 @@ gpg_verify_sync (CamelCipherContext *con
 
 	g_object_unref (filter);
 	g_object_unref (istream);
+	istream = NULL;
 
 	g_seekable_seek (G_SEEKABLE (canon_stream), 0, G_SEEK_SET, NULL, NULL);
 
